Signal/src/ViewControllers/ProfileViewController.m

```diff
@@ -0,0 +1,308 @@
+    [contents addSection:nameSection];
+
+    self.contents = contents;
+}
+
+#pragma mark - Event Handling
+
+- (void)backButtonPressed:(id)sender
+{
+    [self.nameTextField resignFirstResponder];
+
+    if (!self.hasUnsavedChanges) {
+        // If user made no changes, return to conversation settings view.
+        [self.navigationController popViewControllerAnimated:YES];
+        return;
+    }
+
+    UIAlertController *controller = [UIAlertController
+        alertControllerWithTitle:
+            NSLocalizedString(@"NEW_GROUP_VIEW_UNSAVED_CHANGES_TITLE",
+                @"The alert title if user tries to exit the new group view without saving changes.")
+                         message:
+                             NSLocalizedString(@"NEW_GROUP_VIEW_UNSAVED_CHANGES_MESSAGE",
+                                 @"The alert message if user tries to exit the new group view without saving changes.")
+                  preferredStyle:UIAlertControllerStyleAlert];
+    [controller
+        addAction:[UIAlertAction actionWithTitle:NSLocalizedString(@"ALERT_DISCARD_BUTTON",
+                                                     @"The label for the 'discard' button in alerts and action sheets.")
+                                           style:UIAlertActionStyleDestructive
+                                         handler:^(UIAlertAction *action) {
+                                             [self.navigationController popViewControllerAnimated:YES];
+                                         }]];
+    [controller addAction:[UIAlertAction actionWithTitle:NSLocalizedString(@"TXT_CANCEL_TITLE", nil)
+                                                   style:UIAlertActionStyleCancel
+                                                 handler:nil]];
+    [self presentViewController:controller animated:YES completion:nil];
+}
+
+- (void)avatarTapped:(UIGestureRecognizer *)sender
+{
+    if (sender.state == UIGestureRecognizerStateRecognized) {
+        [self.avatarViewHelper showChangeAvatarUI];
+    }
+}
+
+- (void)setHasUnsavedChanges:(BOOL)hasUnsavedChanges
+{
+    _hasUnsavedChanges = hasUnsavedChanges;
+
+    if (hasUnsavedChanges) {
+        self.navigationItem.rightBarButtonItem = (self.hasUnsavedChanges
+                ? [[UIBarButtonItem alloc] initWithTitle:NSLocalizedString(@"EDIT_GROUP_UPDATE_BUTTON",
+                                                             @"The title for the 'update group' button.")
+                                                   style:UIBarButtonItemStylePlain
+                                                  target:self
+                                                  action:@selector(updatePressed)]
+                : nil);
+    }
+}
+
+- (void)updatePressed
+{
+    [self updateProfile];
+}
+
+- (void)updateProfile
+{
+    __weak ProfileViewController *weakSelf = self;
+    [OWSProfilesManager.sharedManager
+         updateLocalProfileName:self.nameTextField.text
```

Comment:
My intent is to eventually validate in both places.  There's a bunch of TODOs in this view and a checklist item here:

https://trello.com/c/TgIe3KQe/1227-add-possibly-1-or-2-views-to-view-update-profile-a-on-launch-b-in-app-settings